packages/cli/internal/pkg/cli/spec/io_test.go

```diff
@@ -352,3 +353,77 @@ workflows:
 		})
 	}
 }
+
+func TestFromJson(t *testing.T) {
```

Comment:
The unmarshalling parts don't seem to be testing the inputs. Maybe consider using real data and do a `reflect.DeepEqual`? We can add this as a TODO.